internal/dag/builder.go

```diff
@@ -38,53 +32,63 @@ type RouteServiceName struct {
 	Port      int32
 }
 
+// Processor uses a DAG builder to construct part
+// of a DAG.
+type Processor interface {
+	// SetBuilder provides the Processor with a Builder
+	// to use.
+	SetBuilder(builder *Builder)
+
+	// Run executes the processor.
+	Run()
```

Comment:
Can we pass the Builder into `Run` and remove `SetBuilder`?

Reply:
Yep, I already ended up doing this in a subsequent WIP commit anyway. I'll make that change for this PR.